File: HackerRank 30 Days/day13/main.cs
using System;

abstract class Book
{
    protected string title;
    protected string author;

    protected Book(string t, string a)
    {
        title = t;
        author = a;
    }

    public abstract void display();
}

class MyBook : Book
{
    private int price;

    public MyBook(string t, string a, int p) : base(t, a)
    {
        price = p;
    }

    public override void display()
    {
        Console.WriteLine("Title: " + title);
        Console.WriteLine("Author: " + author);
        Console.WriteLine("Price: " + price);
    }
}

class Solution
{
    static void Main()
    {
        string title = Console.ReadLine();
        string author = Console.ReadLine();
        int price = int.Parse(Console.ReadLine());

        Book newNovel = new MyBook(title, author, price);
        newNovel.display();
    }
}

File: HackerRank 30 Days/day13/main.js
class Book {
    constructor(title, author) {
        if (this.constructor === Book) {
            throw new TypeError('Cannot instantiate abstract class');
        }
        this.title = title;
        this.author = author;
    }

    display() {
        throw new Error('Abstract method');
    }
}

class MyBook extends Book {
    constructor(title, author, price) {
        super(title, author);
        this.price = price;
    }

    display() {
        console.log(`Title: ${this.title}`);
        console.log(`Author: ${this.author}`);
        console.log(`Price: ${this.price}`);
    }
}

const title = readLine();
const author = readLine();
const price = parseInt(readLine(), 10);

const book = new MyBook(title, author, price);
book.display();
